lambdas/drivingpermitcheck/src/test/java/uk/gov/di/ipv/cri/drivingpermit/api/service/dva/DvaCryptographyServiceTest.java

```diff
@@ -0,0 +1,164 @@
+                                SHA_256_THUMBPRINT + "-encryption"));
+        when(dvaConfiguration.getEncryptionCert())
+                .thenReturn(KeyCertHelper.getDecodedX509Certificate(BASE64_ENCRYPTION_PUBLIC_CERT));
+
+        DvaPayload dvaPayload = createSuccessDvaPayload();
+        JWSObject jwsObject = dvaCryptographyService.preparePayload(dvaPayload);
+
+        JWSHeader jwtHeader = jwsObject.getHeader();
+        Payload payload = jwsObject.getPayload();
+
+        String[] innerJwt = payload.toString().split("\\.");
+
+        Base64.Decoder decoder = Base64.getUrlDecoder();
+
+        String innerJwtHeaderString = new String(decoder.decode(innerJwt[0]));
+        JWEHeader innerJwtHeader = JWEHeader.parse(innerJwtHeaderString);
+
+        String innerPayload = new String(decoder.decode(innerJwt[1]));
+
+        assertEquals(jwtHeader.getCustomParam("x5t"), SHA_1_THUMBPRINT);
+        assertEquals(jwtHeader.getCustomParam("x5t#S256"), SHA_256_THUMBPRINT);
+
+        assertEquals(
+                innerJwtHeader.getX509CertThumbprint().toString(),
+                SHA_1_THUMBPRINT + "-encryption");
+        assertEquals(
+                innerJwtHeader.getX509CertSHA256Thumbprint().toString(),
+                SHA_256_THUMBPRINT + "-encryption");
+
+        assertNotNull(innerPayload);
+    }
+
+    @Test
+    void unwrapDvaResponse()
+            throws CertificateException, JOSEException, JsonProcessingException,
+                    InvalidKeySpecException, NoSuchAlgorithmException, ParseException {
+        when(configurationService.getDvaConfiguration()).thenReturn(dvaConfiguration);
+
+        DvaCryptographyService dvaCryptographyService =
+                new DvaCryptographyService(configurationService);
+
+        when(dvaConfiguration.getSigningKey()).thenReturn(getPrivateKey(BASE64_DCS_SIGNING_KEY));
+        when(dvaConfiguration.getSigningCertThumbprints())
+                .thenReturn(new Thumbprints(SHA_1_THUMBPRINT, SHA_256_THUMBPRINT));
+
+        when(dvaConfiguration.getEncryptionCertThumbprints())
+                .thenReturn(
+                        new Thumbprints(
+                                SHA_1_THUMBPRINT + "-encryption",
+                                SHA_256_THUMBPRINT + "-encryption"));
+        when(dvaConfiguration.getEncryptionCert())
+                .thenReturn(KeyCertHelper.getDecodedX509Certificate(BASE64_ENCRYPTION_PUBLIC_CERT));
+
+        JWSObject jwsResponseObject =
+                dvaCryptographyService.preparePayload(createSuccessDvaResponse());
+
+        when(dvaConfiguration.getSigningCert())
+                .thenReturn(KeyCertHelper.getDecodedX509Certificate(BASE64_DCS_SIGNING_CERT));
+        when(dvaConfiguration.getEncryptionKey())
+                .thenReturn(getPrivateKey(BASE64_ENCRYPTION_PRIVATE_KEY));
+
+        DvaResponse dvaResponse =
+                dvaCryptographyService.unwrapDvaResponse(jwsResponseObject.serialize());
+
+        assertNotNull(dvaResponse);
```

Comment:
Could we also assert some values from the response here just to confirm it hasnt unwrapped an empty object